packages/react-reconciler/src/ReactFiberCacheComponent.new.js

```diff
@@ -200,34 +200,41 @@ export function popRootCachePool(root: FiberRoot, renderLanes: Lanes) {
 
 export function restoreSpawnedCachePool(
   offscreenWorkInProgress: Fiber,
-  prevCachePool: SpawnedCachePool,
+  prevCachePool: SpawnedCachePool | null,
 ): SpawnedCachePool | null {
   if (!enableCache) {
     return (null: any);
   }
-  const nextParentCache = isPrimaryRenderer
-    ? CacheContext._currentValue
-    : CacheContext._currentValue2;
-  if (nextParentCache !== prevCachePool.parent) {
-    // There was a refresh. Don't bother restoring anything since the refresh
-    // will override it.
+
+  if (prevCachePool === null) {
+    push(resumedCache, resumedCache.current, offscreenWorkInProgress);
     return null;
   } else {
-    // No refresh. Resume with the previous cache. New Cache boundaries in the
-    // subtree use this one instead of requesting a fresh one (see
-    // peekCacheFromPool).
-    push(resumedCache, prevCachePool.pool, offscreenWorkInProgress);
-
-    // Return the cache pool to signal that we did in fact push it. We will
-    // assign this to the field on the fiber so we know to pop the context.
-    return prevCachePool;
+    const nextParentCache = isPrimaryRenderer
```

Comment:
This function doesn't need to return anything anymore, right? So this whole `else` block could just be:

```js
push(resumedCache, prevCachePool.pool, offscreenWorkInProgress);
```